server/main.go

```diff
@@ -46,12 +54,18 @@ func main() {
 		cap:   *cap,
 		red:   red,
 		cache: make(map[string]response),
+		dht: &DHT.SimplisticDHT{
```

Comment:
I suggest making a `DHT.New(host)` method, so `DHT.Pair` doesn't have to be exposed.  Note: it should return the interface, not the concrete implementation, along with making the concrete implementation private.